core/src/avm2/globals/flash/display/bitmap_data.rs

```diff
@@ -1122,6 +1122,83 @@ pub fn clone<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implement `BitmapData.paletteMap`
+pub fn palette_map<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(bitmap_data) = this.and_then(|this| this.as_bitmap_data()) {
+        bitmap_data.read().check_valid(activation)?;
+        let source_bitmap = args.get_object(activation, 0, "sourceBitmapData")?
+        .as_bitmap_data()
+        .ok_or_else(|| {
+            Error::from(format!("TypeError: Error #1034: Type Coercion failed: cannot convert {} to flash.display.BitmapData.", args[0].coerce_to_string(activation).unwrap_or_default()))
```

Comment:
I don't think this error is necessary, as it should be automatically coerced to a `BitmapData` in the AS glue.

Reply:
You mean, `unwrap()` will be just enough?